### ISSUE TITLE
Non backward compatible changes for release 4

### ISSUE THREAD

**Issue by sbrunner** [camptocamp/c2cwsgiutils#354]
Use multistage build to remove dev python packages:
```
codacy-coverage
coverage
flake8
junit2html
mypy
pytest
pytest-cov
pytest-html
yamllint
```

**Comment by pvalsecc** [comment]
I don't agree with the removal of python dev packages. Every single child image needs those and will have to install them. And if we remove them, they'll need to install them, copy the source and un-install them. Which means an `apt install` every times the source is touched => Sloooooow...

**Comment by sbrunner** [comment]
Edited => full
